src/Inputs/ComboBox.tsx

```diff
@@ -0,0 +1,241 @@
+    numVisible =NUM_VISIBLE_SELECTIONS,
+    ...props
+}): React.ReactElement => {
+    const [expanded, setExpanded] = useState(false);
+    const [, mount, animation] = useTransition(expanded, {
+        end: theme.speed[SPEED],
+    });
+
+
+
+    const createList = (
+        children: React.ReactNode[],
+        onSelect: React.MouseEventHandler,
+        value?: string | number,
+    ): React.ReactNode[] =>
+        children.map( (child): React.ReactElement | null => {
+
+            if (!myWindow.inputValue){
+                if (child && isValidElement(child)){
+                    const val = child.props.value;
+                    const selected = String(value) === val;
+                    return (
+                        <SelectItem
+                            {...child.props}
+                            active={selected}
+                            onClick={onSelect}
+                            key={val}
+                        />
+                    );}
+            }
+
+            else
+
+            {
+
+            if ((child && isValidElement(child)) && (myWindow.inputValue && (((child.props.children).toLowerCase()).search(myWindow.inputValue.toLowerCase()))> (-1) )) {
+                const val = child.props.value;
+                const selected = String(value) === val;
+                return (
+                    <SelectItem
+                        {...child.props}
+                        active={selected}
+                        onClick={onSelect}
+                        key={val}
+                    />
+                );}
+
+        }
+        
+        return null;} );
+
+    const onSelect = useCallback(
+        ({ currentTarget }): void => {
+            onChange({
+                target: {
+                    value: currentTarget.getAttribute('value'),
+                    name,
+                },
+            });
+            myWindow.inputValue=currentTarget.firstChild.nodeValue;
+        },
+        [name],
+    );
+
+    useEffect((): void | (() => void | undefined) => {
+        isMounted=true;
+
+        if (expanded) {
+            options = Children.toArray(children);
+            const listener = (): void => {
+                if (isMounted){
+                    setExpanded(false);}
+            };
+
+            const timerforkeydown = window.setTimeout((): void => {
+                window.addEventListener('keydown', listener, { once: true });
+            }, 10);
+            const timerforclick = window.setTimeout((): void => {
+                window.addEventListener('click', listener, { once: true });
+            }, 10);
+            return (): void => {
+                isMounted=false;
+                window.clearTimeout(timerforkeydown);
+                window.clearTimeout(timerforclick);
+                window.removeEventListener('keydown', listener);
+                window.removeEventListener('click', listener);
+            };
+        }
+        return undefined;

```

Comment:
The correct return would be return = () => isMounted = false
A useEffect returns a clean up function ( () => {} )

Reply:
so I should remove isMounted=false from line130 , correct?

Reply:
@nikitasheremet mans learning owo

Reply:
@sahandfarahani ur that is correct, but ur implementation of isMounted needs work and should also be set to false on line 137